Examples/Azure/Azure_User_Intune_Device_Apps_Ownership_DeviceApps.ps1

```diff
@@ -0,0 +1,794 @@
+    The id of the device feed to be used.
+
+    .Parameter DWDataTable
+    [System.Data.DataTable] Data table containing the fields required to insert data into the DW Device API.
+
+    .Outputs
+    Output type [string]
+    Text confirming the number of rows to be inserted.
+
+    .Example
+    # Get the device feed id for the named feed.
+    Write-DeviceFeedData -Instance $Instance -DWDataTable $dtDashworksInput -ImportId $DeviceImportID -APIKey $APIKey
+    #>
+
+    [OutputType([string])]
+    Param (
+        [parameter(Mandatory=$True)]
+        [string]$Instance,
+
+        [Parameter(Mandatory=$True)]
+        [System.Data.DataTable]$DWDeviceDataTable,
+
+        [Parameter(Mandatory=$True)]
+        [string]$APIKey,
+
+        [parameter(Mandatory=$False)]
+        [string]$FeedName = $null,
+
+        [parameter(Mandatory=$False)]
+        [string]$ImportId,
+
+        [Parameter(Mandatory=$True)]
+        [System.Data.DataTable]$DWDeviceAppsTable,
+
+        [Parameter(Mandatory=$True)]
+        [string]$AppImportID
+    )
+
+
+    if (-not $ImportId)
+    {
+        if (-not $FeedName)
+        {
+            return 'Device feed not found by name or ID'
+        }
+
+        $ImportId = Get-DwImportDeviceFeed -Instance $Instance -ApiKey $APIKey -Name $FeedName
+
+        if (-not $ImportId)
+        {
+            return 'Device feed not found by name or ID'
+        }
+    }
+
+    $Postheaders = @{
+        "content-type" = "application/json"
+        "X-API-KEY" = "$APIKey"
+    }
+
+    $uri = "{0}/apiv2/imports/devices/{1}/items" -f $Instance, $ImportId
+    $stopwatch =  [system.diagnostics.stopwatch]::StartNew()
+
+    $DWDeviceDataTable | ForEach-Object -Parallel {
+        $Body = $null
+        $Device = $_
+        $Body = $_ | Select-Object *,applications -ExcludeProperty ItemArray, Table, RowError, RowState, HasErrors 
+
+        $applicationsJSON = @()
+        ($using:DWDeviceAppsTable | Where-Object{$_.DeviceId -eq $($Device.uniqueIdentifier)} | Select-Object -Property id,installState) | ForEach-Object {
+            $AppObject = "" | select-object applicationDistHierId,applicationBusinessKey,entitled,installed
+            $AppObject.applicationDistHierId=$using:AppImportID
+            $AppObject.applicationBusinessKey=$_.id
+            if($_.installState -eq "unknown")
+            {
+                $AppObject.installed=$false
+            }else{
+                $AppObject.installed=$true
+            }
+            $AppObject.entitled=$true
+            $applicationsJSON += $AppObject
+        }
+        
+        $Body.applications = $applicationsJSON
+        $Body = $Body | ConvertTo-Json
+
+        Invoke-RestMethod -Headers $using:Postheaders -Uri $using:uri -Method Post -Body $Body | out-null
+    }
+    $stopwatch.Stop()
+    Return "$($DWDeviceDataTable.Rows.Count) devices processed in $($stopwatch.ElapsedMilliseconds)ms"
+}
+
+$ClientId = "c1733be1-4d08-4933-87d0-ba269adbf023"
```

Comment:
Remove all of these variables from the committed version of the script. It looks like they are parameterised at the beginning of the script.